core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -567,6 +568,30 @@ default EventExecutor defaultSubscriberExecutor() {
         return FuseableStreamMessage.error(this, function);
     }
 
+    /**
+     * Peeks values emitted by this {@link StreamMessage} and applies the specified {@link Consumer}.
+     *
+     * <p>For example:<pre>{@code
+     * StreamMessage<Integer> source = StreamMessage.of(1, 2, 3, 4, 5);
+     * StreamMessage<Integer> ifEvenExistsThenThrow = source.peek(x -> {
+     *      if (x % 2 == 0) {
+     *          throw new IllegalArgumentException();
+     *      }
+     * });
+     * }</pre>
+     */
+    default StreamMessage<T> peek(Consumer<? super T> action, Class<? extends T> type) {
+        requireNonNull(action, "action");
+        requireNonNull(type, "type");
+        final Function<T, T> function = obj -> {
+            if (type.isInstance(obj)) {
+                action.accept(obj);
+            }
+            return obj;
+        };
+        return map(function);
```

Comment:
✅ Based on this [comment](https://github.com/line/armeria/pull/3949#discussion_r757194405), I delegate to `map()` instead of adding new factory method on `FuseableStreaMessage`

Reply:
nit, `peek(Consumer<> action, Class<> type)` vs `peek(Class<> type, Consumer<> action)`? 🤔

Reply:
+1 for `peek(Consumer<> action, Class<> type)`

and how about adding back `peek(Consumer<> action)` so that users don't have to specify the class type when it produces only one type?

Reply:
What I meant was:
```java
default StreamMessage<T> peek(Consumer<? super T> action) {...}
```
So that a user just can
```java
StreamMessage<Integer> ifEvenExistsThenThrow = source.peek(x -> {
     if (x % 2 == 0) {
         throw new IllegalArgumentException();
     }
}); // Integer.class is not specified.
```
I borrow it from your example. 😆